library/Asana/Api/Task.hs

```diff
@@ -47,22 +48,44 @@ import RIO.Time
 
 -- | Just what we need out of our @custom_fields@ for cost and carry-over
 data CustomField
-  = CustomNumber Text (Maybe Integer)
-  | CustomEnum Text (Maybe Text)
+  = CustomNumber Gid Text (Maybe Integer)
+  | CustomEnum Gid Text [EnumOption] (Maybe Text)
   | Other -- ^ Unexpected types dumped here
   deriving (Eq, Generic, Show)
 
+data EnumOption = EnumOption
+  { eoGid :: Gid
+  , eoName :: Text
+  }
+  deriving (Eq, Generic, Show)
+
+instance FromJSON EnumOption where
+  parseJSON = genericParseJSON $ aesonPrefix snakeCase
+
+customEnumId :: CustomField -> Maybe Gid
+customEnumId (CustomEnum _ name opts _) =
+  eoGid <$> find ((== name) . eoName) opts
```

Comment:
This is a bug, it's finding an enum option with the same name as the field name, not the value.